src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21858.cs

```diff
@@ -0,0 +1,25 @@
+﻿using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue21858 : _IssuesUITest
+{
+
+	public Issue21858(TestDevice testDevice) : base(testDevice)
+	{
+	}
+
+	public override string Issue => "DatePicker not respecting UserAppTheme";
+
+	[Test]
+	[Category(UITestCategories.DatePicker)]
+	public void DatePickerShouldRespectUserAppTheme()
+	{
+		App.WaitForElement("ToggleThemeButton");
+		App.Click("ToggleThemeButton");
+		App.Click("DatePicker");
+		VerifyScreenshot();
```

Comment:
Pending snapshots, already available in the latest build.

Example, iOS:
![image](https://github.com/user-attachments/assets/749a2bf6-c55f-4354-b72b-1bde525a5a93)

Could you commit the images?